### PR TITLE
Refactor logger

### PR DESCRIPTION
**Pull request by ElecTwix**
fix #91

### CODE REVIEW

**Comment by ElecTwix** [comment]
```
?       github.com/surrealdb/surrealdb.go/internal/mock [no test files]
?       github.com/surrealdb/surrealdb.go/internal/rpc  [no test files]
?       github.com/surrealdb/surrealdb.go/pkg/gorilla   [no test files]
?       github.com/surrealdb/surrealdb.go/pkg/rand      [no test files]
?       github.com/surrealdb/surrealdb.go/pkg/websocket [no test files]
ok      github.com/surrealdb/surrealdb.go       0.354s
ok      github.com/surrealdb/surrealdb.go/internal/benchmark    (cached) [no tests to run]
ok      github.com/surrealdb/surrealdb.go/pkg/logger    0.002s
```

Seems like something is wrong with CI, I tested with two different machines.

**Comment by ElecTwix** [comment]
waiting for #88

**Comment by ElecTwix** [comment]
for information: this PR removed 1.20 golang version for slog

File: pkg/logger/slog/slog_test.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+		{fn: logger.Error, level: rawslog.LevelError},
+		{fn: logger.Warn, level: rawslog.LevelWarn},
+		{fn: logger.Info, level: rawslog.LevelInfo},
+		{fn: logger.Debug, level: rawslog.LevelDebug},
+	}
+
+	for _, v := range testMethods {
+		t.Run(fmt.Sprintf("testing %s", v.level.String()), func(t *testing.T) {
+			err := checkMethod(v.fn, buffer, v.level.String())
+			if err != nil {
```
Comment (phughk) [inline pkg/logger/slog/slog_test.go:51]: this can be `t.Require().NoError(err)` I think

**Review by phughk — COMMENTED** [review]
Minor comment but otherwise lgtm thank you :) Will approve once resolved

File: pkg/logger/slog/slog_test.go (hunk 1 above)
Comment (ElecTwix) [reply pkg/logger/slog/slog_test.go]: done

**Review by phughk — APPROVED** [review]
Lgtm thank you!

**Pull request merged**